README.md

```diff
@@ -110,10 +140,14 @@ calls `Conn.generatePackets()` and sends the result to the proxyClient specified
 ### `Conn.attach()`
 
 ```ts
-Conn.attach(pclient: Client)
+Conn.attach(pclient: Client, options?: { toClientMiddleware?: PacketMiddleware[], toServerMiddleware?: PacketMiddleware[] })
 ```
+* pclient - The client to be attached
+* options - Optional
+  * toClientMiddleware - Optional. A middleware function array to be used as this clients middle ware to the client. See middleware for a function definition.
+  * toServerMiddleware - Optional. A middleware function array to be used as this clients middle ware to the server
 
-the pclient specified will be added to the `Conn.pclients` array, which means that it will receive all packets from the server. If you want the client to be able to send packets to the server as well, don't forget to call `Conn.link()`
+the pclient specified will be added to the `Conn.pclients` array and the `Conn.receivingPclients`, which means that it will receive all packets from the server. If you want the client to be able to send packets to the server as well, don't forget to call `Conn.link()`
```

Comment:
```suggestion
the pclient specified will be added to the `Conn.pclients` array and the `Conn.receivingPClients`, which means that it will receive all packets from the server. If you want the client to be able to send packets to the server as well, don't forget to call `Conn.link()`
```